Mathlib/ModelTheory/Fraisse.lean

```diff
@@ -403,6 +407,42 @@ theorem nonempty_equiv : Nonempty (M ≃[L] N) := by
 
 end IsFraisseLimit
 
+namespace empty
+
+/-- Any countable infinite structure in the empty language is a Fraïssé limit of the class of finite
+structures. -/
+theorem isFraisseLimit_of_countable_infinite
+    (M : Type*) [Countable M] [Infinite M] [Language.empty.Structure M] :
+    IsFraisseLimit { S : Bundled Language.empty.Structure | Finite S } M where
+  age := by
+    ext S
+    simp only [age, Structure.fg_iff_finite, empty.nonempty_embedding_iff, Cardinal.lift_id,
+      Cardinal.mk_eq_aleph0, mem_setOf_eq, and_iff_left_iff_imp]
+    exact fun hS => Cardinal.mk_le_aleph0
+  ultrahomogeneous := by
+    intro S hS f
+    haveI : Finite S := hS.finite
```

Comment:
Is there a reason these need to be `haveI` rather than `have`?

Reply:
The only reason is that I haven't fully converted from Lean3 syntax. Should I also change this elsewhere in the file, or will that make the diff too weird?

Reply:
I'd leave it for now!